app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def filter(filtering_params)
+      results = self.where(nil)
+      filtering_params.each do |key, value|
+        results = results.public_send(key, value) if value.present?
+      end
```

Comment:
I personally don’t think we should be accepting every key/value pair as a valid method/input. It does make the code DRY but it is hard predict the behaviour. Could we implement a whitelist perhaps to limit the accepted parameters?